apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts

```diff
@@ -329,6 +392,18 @@ export class BulkScriptAction extends BaseScriptAction {
         ' For details, refer to the website article "Selecting subsets of projects".'
     });
 
+    this._watchParameter = this.defineFlagParameter({
+      parameterLongName: '--watch',
+      parameterShortName: '-w',
+      description:
+        'Normally Rush would terminate after the command finishes;' +
+        ' adding this parameter will instead watch the file system for changes to the selected projects' +
+        ' (or all projects if no selection was specified).' +
+        ' When changes are detected, will re-execute the command on all projects within the selection that are' +
+        ' impacted by the detected changes, then resume waiting for changes.' +
+        ' For details, refer to the website article "Using watch mode".'
+    });
+
```

Comment:
In my testing, I found that if the initial invocation of `rush build --watch -T my-app` encounters 1 warning, then the operation will terminate immediately instead of entering a watch loop.

Reply:
And if it successfully enters a watch loop, 1 warning seems sufficient to break out of the loop.  This is inconsistent with how `heft build --watch` works.

Reply:
Updated to have any errors from the underlying build get swallowed by the watcher.

Reply:
This is still reproing.  Here's my callstack:
```
Error: An error occurred.
    at BulkScriptAction._runOnce (C:\Git\rushstack\apps\rush-lib\lib\cli\scriptActions\BulkScriptAction.js:382:19)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async BulkScriptAction.runAsync (C:\Git\rushstack\apps\rush-lib\lib\cli\scriptActions\BulkScriptAction.js:151:13)
    at async RushCommandLineParser.onExecute (C:\Git\rushstack\libraries\ts-command-line\lib\providers\CommandLineParser.js:199:13)
    at async RushCommandLineParser._wrapOnExecuteAsync (C:\Git\rushstack\apps\rush-lib\lib\cli\RushCommandLineParser.js:140:9)
    at async RushCommandLineParser.onExecute (C:\Git\rushstack\apps\rush-lib\lib\cli\RushCommandLineParser.js:122:13)
    at async RushCommandLineParser.execute (C:\Git\rushstack\libraries\ts-command-line\lib\providers\CommandLineParser.js:100:13)
```

It's going through [this call](https://github.com/microsoft/rushstack/blob/5207e98233858781c7ca1c3d9b4398c6008e370f/apps/rush-lib/src/cli/scriptActions/BulkScriptAction.ts#L207) whereas the `try`/`catch` block you added is in `runWatch()`.

Reply:
This calls stack isn't in the code path for `--watch`.